scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,209 @@
+
+  private List<ImageFrame> imageFrames;
+  private ImageOverlay imageOverlay;
+
+  private Integer frameIndex = 0;
+  private Integer period = 67;
+  private Timer timer;
+  private boolean isTimerRunning;
+
+  public void initialize() {
+
+    try {
+      
+      // populate the frames combo box with values
+      framesComboBox.getItems().addAll("60 frames per second", "30 frames per second", "15 frames per second");
+      // open the sample at 15fps
+      framesComboBox.getSelectionModel().select(2);
+      // set timer running tracker to true when sample loads
+      isTimerRunning = true;
+
+      // create a new ArcGISScene and set it to the scene view
+      ArcGISScene scene = new ArcGISScene();
+      sceneView.setArcGISScene(scene);
+
+      // create a camera, looking at the pacific southwest sector
+      Point observationPoint = new Point(-116.621, 24.7773, 856977.0);
+      Camera camera = new Camera(observationPoint, 353.994, 48.5495, 0.0);
+
+      // create an envelope of the pacific southwest sector for displaying the image frame
+      Point pointForImageFrame = new Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84());
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
+
+      // create a new tiled layer from the World Dark Gray Base REST service and set it as the scene's basemap
+      Basemap basemap = new Basemap(new ArcGISTiledLayer("https://services.arcgisonline" +
+        ".com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"));
+      scene.setBasemap(basemap);
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+      // store the newly created image overlay
+      imageOverlay = sceneView.getImageOverlays().get(0);
+
+      // instantiate a new empty list to hold image frames
+      imageFrames = new ArrayList<>();
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files by file name in ascending order
+      if (imageFiles != null) {
+        Arrays.sort(imageFiles);
+        // create an image frame from the file path and add it to the list of image frames
+        for (File file : imageFiles) {
+          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
+          imageFrames.add(imageFrame);
```

Comment:
@Rachael-E regarding the performance aspect of this, have you tried holding on to an `ArrayList<Image>` and creating the `ImageFrame` ad-hoc, only when it needs to be displayed? There seems to be quite a difference in memory consumption between a collection of `Image` and `ImageFrame`, as other platforms are reporting

Reply:
thanks @JonLavi no haven't tried this: is this the agreed way forward from the samples team perspective? I've been out the loop :D just waiting on @tschie 2nd review and I can make changes then: I know we're already going to change the drop down text from "xx fps" to "slow" etc. If there's anything else to go in I'd be grateful to hear it!

Reply:
There's no concensus as of yet for which way is to be chosen across all platforms. It looks like for some, an `ArrayList<Image>` provies a less smooth animation. I think it would be a matter of testing it out and seeing whether it runs smoothly still and consumes less memory. Hopefully the team can agree to add some info to the 'Additional Information' section of the Readme - it's still in PR.